app/components/ITunesCard/index.js

```diff
@@ -0,0 +1,132 @@
+/**
+ *
+ * ITunesCard
+ *
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import truncate from 'lodash/truncate';
+import { isEmpty } from 'lodash';
+import { Card, Popover, Button, Progress } from 'antd';
+import { PlayCircleOutlined, PauseCircleOutlined } from '@ant-design/icons';
+import T from '@components/T';
+import If from '@components/If';
+import { colors } from '@app/themes';
+import { ACTIONS } from '@utils/constants';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 1.25em 0;
+    box-shadow: ${colors.boxShadow};
+  }
+`;
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-bottom: 0.75rem;
+  gap: 1em;
+`;
+const CustomImg = styled.img`
+  display: block;
+  margin: 0.75rem auto;
+  height: 10rem;
+`;
+
+export function ITunesCard({ track, onActionClick }) {
+  const audioRef = useRef(null);
+  const [progress, setProgress] = useState(0);
+  const [intervalStore, setIntervalStore] = useState(null);
+
+  useEffect(() => {
+    clearInterval(intervalStore);
+  }, [track]);
+
+  function calculateProgress() {
+    const intervalValue = setInterval(() => {
+      const fraction = (audioRef.current.currentTime / audioRef.current.duration) * 100;
+      setProgress(fraction);
+    }, 1000);
+    setIntervalStore(intervalValue);
+  }
+  const handleTrackPlay = (action) => {
+    if (action === ACTIONS.PLAY) {
+      audioRef.current.play();
+      onActionClick(audioRef);
+      calculateProgress();
+    }
+    if (action === ACTIONS.PAUSE) {
+      audioRef.current.pause();
+      clearInterval(intervalStore);
```

Comment:
we need to always propagate the onActionClick. We should pass the `action` as well so that the consuming client can determine whether play/pause was clicked